com.unity.sg2/Editor/GraphUI/GraphElements/ModelUI/TransformDropdownsPart.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using Unity.GraphToolsFoundation.Editor;
+using UnityEngine.UIElements;
+
+namespace UnityEditor.ShaderGraph.GraphUI
+{
+    using CoordinateSpace = UnityEditor.ShaderGraph.GraphDelta.CoordinateSpace;
+    using ConversionType = UnityEditor.ShaderGraph.GraphDelta.ConversionType;
+
+    class TransformDropdownsPart : BaseModelViewPart
+    {
+        VisualElement m_Root;
+        EnumField m_FromDropdown, m_ToDropdown;
+        EnumField m_ModeDropdown;
+        Label m_Label;
+
+        public override VisualElement Root => m_Root;
+
+        public TransformDropdownsPart(string name, Model model, ModelView ownerElement, string parentClassName)
+            : base(name, model, ownerElement, parentClassName) { }
+
+        protected override void BuildPartUI(VisualElement parent)
+        {
+            if (m_Model is not GraphDataNodeModel sgNodeModel) return;
+
+            EventCallback<ChangeEvent<Enum>> SetCoordinateSpaceCallback(string fieldName)
+            {
+                return e =>
+                {
+                    m_OwnerElement.RootView.Dispatch(new SetCoordinateSpaceCommand(sgNodeModel, fieldName, (CoordinateSpace)e.newValue));
+                };
+            }
+
+            m_Root = new VisualElement();
+
+            GraphElementHelper.LoadTemplateAndStylesheet(m_Root, "TransformDropdownsPart", "sg-transform-dropdowns");
+
+            m_FromDropdown = m_Root.Q<EnumField>("from");
+            m_FromDropdown.Init(default(CoordinateSpace));
+            m_FromDropdown.RegisterValueChangedCallback(SetCoordinateSpaceCallback(GraphDelta.TransformNode.kSourceSpace));
```

Comment:
Something I learned from the memory leak stuff I fixed in SG1 is that we need to have an explicit step in our application lifecycle to release resources and unbind callbacks, which is something to think about as we add any callbacks going forwards

Not a blocker for this PR, as I think we need a more general purpose solution, but wanted to put that on your radar